snippets/terraform/deployments/with-diagnostic-setting/main.tf

```diff
@@ -0,0 +1,57 @@
+terraform {
+  required_version = "~> 1.3"
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~> 3.85"
+    }
+  }
+}
+
+provider "azurerm" {
+  features {}
+}
+
+module "prerequisites" {
+  source   = "../../prerequisites"
+  location = var.location
+  name     = var.name
+  tags     = var.tags
+}
+
+data "azurerm_storage_account" "example" {
+  name                = var.storage_account_name
+  resource_group_name = var.storage_account_resource_group
+}
+
+resource "azurerm_nginx_deployment" "example" {
+  name                     = var.name
+  resource_group_name      = module.prerequisites.resource_group_name
+  sku                      = var.sku
+  location                 = var.location
+  capacity                 = 50
+  diagnose_support_enabled = false
+
+  identity {
```

Comment:
a comment here stating that it will only work with system assigned might be nice.